File: platform-mobile/src/main/java/ru/pocketbyte/locolaser/platform/mobile/IosPlistPlatformConfig.java
package ru.pocketbyte.locolaser.platform.mobile;

import ru.pocketbyte.locolaser.config.platform.BasePlatformConfig;
import ru.pocketbyte.locolaser.platform.mobile.resource.IosPlistResources;
import ru.pocketbyte.locolaser.resource.PlatformResources;

/**
 * iOS Info.plist localisation configuration.
 *
 * @author Denis Shurygin
 */
public class IosPlistPlatformConfig extends BasePlatformConfig {
    public static final String TYPE = "ios_plist";

    @Override
    public String getType() {
        return TYPE;
    }

    @Override
    public String getDefaultTempDirPath() {
        return "../DerivedData/LocoLaserTemp/";
    }

    @Override
    public String getDefaultResourcesPath() {
        return "./";
    }

    @Override
    protected String getDefaultResourceName() {
        return "InfoPlist";
    }

    @Override
    public PlatformResources getResources() {
        return new IosPlistResources(getResourcesDir(), getResourceName());
    }

}

File: platform-mobile/src/main/java/ru/pocketbyte/locolaser/platform/mobile/IosPlistPlatformConfig.kt
package ru.pocketbyte.locolaser.platform.mobile

import ru.pocketbyte.locolaser.config.platform.BasePlatformConfig
import ru.pocketbyte.locolaser.platform.mobile.resource.IosPlistResources

/**
 * iOS Info.plist localisation configuration.
 *
 * @author Denis Shurygin
 */
class IosPlistPlatformConfig : BasePlatformConfig() {

    companion object {
        const val TYPE = "ios_plist"
    }

    override val type = TYPE

    override val defaultTempDirPath = "../DerivedData/LocoLaserTemp/"
    override val defaultResourcesPath = "./"
    override val defaultResourceName = "InfoPlist"

    override val resources
        get() = IosPlistResources(resourcesDir!!, resourceName!!)

}